docs/designs/013-audit-logging.md

```diff
@@ -0,0 +1,735 @@
+// CR name format: maintenance-<nodeName>-<healthEventID>
+// Example: maintenance-gpu-node-01-abc123def456
+func extractHealthEventID(resourceInfo ResourceInfo) string {
+    if !strings.HasPrefix(resourceInfo.Name, "maintenance-") {
+        return ""
+    }
+    
+    // Split: ["maintenance", "gpu", "node", "01", "abc123def456"]
+    parts := strings.Split(resourceInfo.Name, "-")
+    if len(parts) >= 3 {
+        return parts[len(parts)-1]  // Last part is HealthEventID
+    }
+    return ""
+}
+```
+
+### Integration Pattern
+
+Components wrap their Kubernetes client with the auditing transport:
+
+```go
+// In component initialization - ONE place, not everywhere
+func NewK8sClient(ctx context.Context, component auditlog.ComponentType) (*Client, error) {
+    // Create audit logger with configured sinks
+    auditLogger := auditlog.NewLogger(ctx, auditlog.ConfigFromEnv(component))
+    auditLogger.Start()
+    
+    // Get base REST config
+    restConfig, err := rest.InClusterConfig()
+    if err != nil {
+        return nil, err
+    }
+    
+    // Wrap transport with auditing round-tripper
+    restConfig.Wrap(func(rt http.RoundTripper) http.RoundTripper {
+        return auditlog.NewAuditingRoundTripper(rt, auditLogger, component)
+    })
+    
+    // Create clientset with auditing transport
+    clientset, err := kubernetes.NewForConfig(restConfig)
+    if err != nil {
+        return nil, err
+    }
+    
+    return &Client{
+        clientset:   clientset,
+        auditLogger: auditLogger,
+    }, nil
+}
+
+// No audit code needed in operation methods!
+func (c *Client) QuarantineNode(nodeName string) error {
+    // Just perform the operation - auditing happens automatically
+    _, err := c.clientset.CoreV1().Nodes().Patch(
+        context.Background(),
+        nodeName,
+        types.StrategicMergePatchType,
+        patchBytes,
+        metav1.PatchOptions{},
+    )
+    return err
+    // Audit log is automatically captured by the RoundTripper
+}
+```
+
+### CSP SDK Integration
+
+The same RoundTripper pattern works for all Cloud Service Provider SDKs:
+
+#### AWS SDK v2
+```go
+func NewAWSClient(ctx context.Context, logger *auditlog.Logger) (*ec2.Client, error) {
+    auditingHTTPClient := &http.Client{
+        Transport: auditlog.NewAuditingRoundTripper(http.DefaultTransport, logger, "janitor"),
+    }
+
+    cfg, err := config.LoadDefaultConfig(ctx,
+        config.WithRegion(os.Getenv("AWS_REGION")),
+        config.WithHTTPClient(auditingHTTPClient),  // Inject auditing client
+    )
+    if err != nil {
+        return nil, err
+    }
+
+    return ec2.NewFromConfig(cfg), nil
+}
+```
+
+#### GCP SDK
+```go
+func NewGCPClient(ctx context.Context, logger *auditlog.Logger) (*compute.InstancesClient, error) {
+    auditingHTTPClient := &http.Client{
+        Transport: auditlog.NewAuditingRoundTripper(http.DefaultTransport, logger, "janitor"),
+    }
+
+    client, err := compute.NewInstancesRESTClient(ctx,
```

Comment:
Doesn't GCP SDK also offer protobuf based clients? What would we do in those cases?